mysql/mysqldump.go

```diff
@@ -0,0 +1,578 @@
+	}
+}
+
+// getTableName extracts the table name from *ast.TableName table, and returns
+// the raw extracted name (the MySQL table name).
+// *ast.TableName is used to represent table names. It consists of two components:
+//  Schema: schemas are MySQL db often unspecified;
+//  Name: name of the table
+// We build a table name from these components as follows:
+// a) nil components are dropped.
+// b) if more than one component is specified, they are joined using "."
+//    (Note that Spanner doesn't allow "." in table names, so this
+//    will eventually get re-mapped when we construct the Spanner table name).
+// c) return error if Table is nil or "".
+func getTableName(table *ast.TableName) (string, error) {
+	var l []string
+
+	if table.Schema.String() != "" {
+		l = append(l, table.Schema.String())
+	}
+	if table.Name.String() == "" {
+		return "", fmt.Errorf("tablename is empty: can't build table name")
+	}
+	l = append(l, table.Name.String())
+	return strings.Join(l, "."), nil
+}
+
+func processColumn(conv *internal.Conv, tableName string, col *ast.ColumnDef) (string, schema.Column, bool, error) {
+	if col.Name == nil {
+		return "", schema.Column{}, false, fmt.Errorf("column name is nil")
+	}
+	name := col.Name.OrigColName()
+	if col.Tp == nil {
+		return "", schema.Column{}, false, fmt.Errorf("can't get column type for %s: %w", name, fmt.Errorf("found nil *ast.ColumnDef.Tp"))
+	}
+	tid, mods := getTypeModsAndID(col.Tp.String())
+	ty := schema.Type{
+		Name:        tid,
+		Mods:        mods,
+		ArrayBounds: getArrayBounds(col.Tp.String(), col.Tp.Elems)}
+	column := schema.Column{Name: name, Type: ty}
+	return name, column, updateColsByOption(conv, tableName, col, &column), nil
+}
+
+// updateColsByOption is specifially for ColDef constraints.
+// ColumnOption type is used for parsing column constraint info from MySQL
+func updateColsByOption(conv *internal.Conv, tableName string, col *ast.ColumnDef, column *schema.Column) bool {
+	isPk := false
+	for _, elem := range col.Options {
+		switch op := elem.Tp; op {
+		case ast.ColumnOptionPrimaryKey:
+			column.NotNull = true
+			// If primary key is defined in a column then `isPk` will be true
+			// and this column will be added in colDef as primary keys
+			isPk = true
+		case ast.ColumnOptionNotNull:
+			column.NotNull = true
+		case ast.ColumnOptionDefaultValue:
+			column.Ignored.Default = true
+		case ast.ColumnOptionUniqKey:
+			column.Unique = true
+		case ast.ColumnOptionCheck:
+			column.Ignored.Check = true
+		case ast.ColumnOptionReference:
+			column.Ignored.ForeignKey = true
+		default:
+			conv.Unexpected(fmt.Sprintf("Not supported type option of const value `%v` found in table : %s , column : %s. Mapping of columnOptionType can be found here: 'https://godoc.org/github.com/pingcap/parser/ast#ColumnOptionType'", op, tableName, column.Name))
+		}
+	}
+	return isPk
+}
+
+// getTypeModsAndID returns ID and mods of column datatype
```

Comment:
Sentences should end in '.'
Same comment at l356 and l358 and l363.
Maybe double check all comments?

Reply:
Done. Also, double checked other comments and made sure it ends with a '.'